### PR TITLE
feat(dRICH): assemble sensors in Photodetector Units (PDUs)

### PR DESCRIPTION
**Pull request by JunhuaiXu**
### Briefly, what does this PR introduce?

This is a big change for the dRICH sensor geometry. It was initially started by @JunhuaiXu and continued by @c-dilks with the advice of Marco Contalbrigo, et al. 

#### Summary of changes:
- adjust some thicknesses, such as vessel walls and acrylic filter
- refactor of the sensor geometry (which on `main` is just a thin `Box` for each SiPM)
  - add the resin backing, following [the S13361-3050NE-08 SiPM datasheet](https://www.hamamatsu.com/eu/en/product/optical-sensors/mppc/mppc_mppc-array/S13361-3050NE-08.html); the photosensitive volume plus the resin backing forms a "SiPM assembly"
  - make a 2x2 matrix of SiPM assemblies, forming a Photodetector Unit (PDU); include service materials behind each 2x2 matrix forming the full PDU assembly
    - `frontservices`, currently just a box
    - `boards`: 5 circuit boards, coming out of the `frontservices` box
    - `backservices`, currently just another box
    - all of these are made of Aluminum, for now
  - The spherical tiling algorithm now tiles PDU assemblies, instead of the original SiPM `Box`es
- Change the readout bit fields, so sensors are identified by a PDU number `pdu` and a SiPM number (0-3) `sipm`, instead of just a single `module` number
- Extrude the annular region of the vessel (the part outside the aerogel snout) toward the -z direction, to allow the sensor services to fit; this is a step toward modelling sensor boxes, which will house each sector's set of PDUs (outside the scope of this PR)

#### Pictures and Geometry details:
See `sensors.pdf` from <https://indico.bnl.gov/event/19848/>

---

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature, closing the following:
  - close https://github.com/eic/epic/issues/111
  - close https://github.com/eic/epic/issues/175
  - close https://github.com/eic/epic/issues/333
  - close https://github.com/eic/epic/issues/328
- [ ] Documentation update
- [ ] Other: __

---

### Please check if this PR fulfills the following:
- [x] Tests for the changes have been added
- [X] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

---

### Does this PR introduce breaking changes? What changes might users need to make to their code?
**YES**, addressed by these PRs; all of them should be merged simultaneously:
- https://github.com/eic/EICrecon/pull/727
- https://github.com/eic/drich-dev/pull/83

#### Things to check downstream:
- [x] pixel gap cuts
- [x] event display and readout
- [x] are the photons stopped by the resin, or do they still penetrate through?
- [x] sensor thickness and IRT geometry conversion
- [x] number of photons / interaction of photons at the resin/sensor/air surface
- [x] PID

---

### Does this PR change default behavior?
yes, photons will stop within the resin volume (currently on `main` they pass through and stop at the vessel wall behind the sensors); this should also prevent photons causing hits from behind (unlikely, but can happen) and from the sensor sides (because of the resin walls)

### CODE REVIEW

**Comment by c-dilks** [comment]
Hmm... somehow this looks like it's based off an older commit on the `main` branch, but Github tells me it's indeed based off the most recent commit on `main`. In other words, it's not possible to update your branch with respect to `main` since Github thinks it is already as up-to-date as possible. Let me see if I can fix it with `git rebase` or something...

**Comment by c-dilks** [comment]
Fixed! I created temporary branch `before-rebase` to backup what you had before the fix, in case anything got lost in the rebase.

**Comment by c-dilks** [comment]
If I understand correctly, the full sensor thickness should be 1.35mm, all resin. The first 0.1mm of that is the sensitive part (currently we use `AirOptical` as the material, for technical reasons), so we should modify this to place the sensitive volume inside a resin volume, rather than two separate volumes.

With this change, we will have to check the number of photons carefully, so perhaps we delay merging this PR until then.

2022-12-05-122925_741x554_scrot

**Comment by c-dilks** [comment]
Full schematic:
2023-02-28-122444_1276x623_scrot

**Comment by c-dilks** [comment]
This is starting to look good, however, we need to check things carefully downstream (checklist added to PR description above).

Here is a picture (grey=resin, blue=sensor active area)
2023-02-28-151155_501x558_scrot

File: src/DRICH_geo.cpp [hunk 1]
```diff
@@ -37,90 +37,105 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
 
   // attributes, from compact file =============================================
```
Comment (c-dilks) [inline src/DRICH_geo.cpp:38]: The majority of changes in this following block are `double` -> `auto` and formatting changes. There are some new variables though, for the PDU and sensor geometry.

**Comment by c-dilks** [comment]
@chchatte92 have you tested this yet? Shall we open this PR for review?

**Comment by wdconinc** [comment]
Quick questions: 
- Is there a way adjust visualization settings so the services behind the SiPMs are not shown?
- Will the same PDUs be used by pfRICH? Does it make sense to start extracting them as a separate source code unit (header + implementation) with the eye on future reuse?

**Comment by chchatte92** [comment]
@wdconinc I guess the PDU will be different for pfRICH.

**Comment by wdconinc** [comment]
> changing a true to a false somewhere

Yeah, usually a `showdaughters=false` is sufficient.

> Somewhat related: root-project/jsroot

Our geoviewer, https://github.com/eic/epic/blob/main/_layouts/geoviewer.html, live at https://eic.github.io/epic/geoviewer, seems to be up to date.

> can provide the exact numbers

Ok, thanks for the info. No numbers needed, or at least not in this PR.

**Comment by wdconinc** [comment]
> linux-eic-shell / dawn-view (view1, epic_brycecanyon) (pull_request) Successful in 2m

As long as this doesn't change, the 2 minutes are fine and we don't need to modify the visualization, I think.

> linux-eic-shell / check-overlap-geant4 (epic_brycecanyon) (pull_request) Successful in 69m

This always takes long, and probably isn't your fault anyway :-D

File: src/DRICH_geo.cpp [hunk 2]
```diff
@@ -494,63 +502,297 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
...
+           *       ||  ||  ||  ||  ||
+           *       ||  ||  ||  ||  || front-end and
+           *       ||  ||  ||  ||  || readout boards
+           *       ||  ||  ||  ||  ||
+           *       ||  ||  ||  ||  ||
+           *               ||
+           *               ||
+           *       +----------------+
+           *       | Sockets, etc.  |  backservices
+           *       +----------------+
```
Comment (wdconinc) [inline src/DRICH_geo.cpp:565]: May also be useful in xml, though I understand the desire to avoid duplication.

**Review by wdconinc — APPROVED** [review]
This looks nice, and adds a great amount of detail!

**Comment by alexander-kiselev** [comment]
Hi Chris,

  no, we don't. pfRICH sensor layout s completely different.

  Cheers,
    Alexander.

________________________________

File: src/DRICH_geo.cpp (hunk 2 above)
Comment (c-dilks) [reply src/DRICH_geo.cpp]: Moved these diagrams to the xml in 1e43f01.

**Comment by wdconinc** [comment]
> This should be merged at the same time as https://github.com/eic/EICrecon/pull/727.

Right! thanks :)

**Pull request merged**